pkg/api/base.go

```diff
@@ -15,13 +15,19 @@
 package api
 
 import (
+	"encoding/json"
+	"net/http"
 	"net/url"
+	"reflect"
+	"strings"
 
 	"github.com/astaxie/beego"
+	"github.com/opensds/opensds/pkg/model"
 )
 
 type BasePortal struct {
 	beego.Controller
+	Self interface{}
```

Comment:
Can you explain why add this filed?

Reply:
This variable is used in action method, so that the it can invoke its children's method.